### PR TITLE
Changed the data flow of string type data

### PR DESCRIPTION
**Pull request by muskankhedia**
In this PR, the data flow of the string type data is changed using pointers. And the test cases for the plain.go has been modified.

### CODE REVIEW

**Comment by Harkishen-Singh** [comment]
please change ```./gobase``` to the complete address in maintut.go

**Comment by muskankhedia** [comment]
Yaa..sorry

**Comment by Harkishen-Singh** [comment]
Please fix the pipeline errors. They can be fixed. Normal syntax errors

File: gobase/intermediate_test.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+)
+
+type SaveArrResult struct{
+	path string
+	data []string
+	expected bool
+	checkeddata string
+} 
+
+var SaveArrResults = []SaveArrResult{
```
Comment (Harkishen-Singh) [inline gobase/intermediate_test.go]: indent spaces

File: gobase/plain_test.go [hunk 2]
```diff
@@ -9,46 +9,29 @@ type plainContainer struct {
...
-
-	// save operations
-
-	Save(test1.location, test1.value)
-	Save(test2.location, test2.value)
-	Save(test3.location, test3.value)
-	Save(test4.location, test4.value)
-
-	// retrive operations
+	var plainTests = []plainContainer{
```
Comment (Harkishen-Singh) [inline gobase/plain_test.go]: indent space

File: gobase/plain_test.go [hunk 3]
```diff
@@ -9,46 +9,29 @@ type plainContainer struct {
...
+		{"testPlain2", "@#$%#@(*&^%$)", "", false,"string",""},
+	}
 
-	test1.retr, test1.status = Retrive(test1.location)
-	test2.retr, test2.status = Retrive(test2.location)
-	test3.retr, test3.status = Retrive(test3.location)
-	test4.retr, test4.status = Retrive(test4.location)
 
-	// checks
+	for _,ele := range plainTests{
```
Comment (Harkishen-Singh) [inline gobase/plain_test.go]: indent space

File: gobase/plain_test.go [hunk 4]
```diff
@@ -9,46 +9,29 @@ type plainContainer struct {
...
-	if !(test1.value == test1.retr) || !test1.status {
-		t.Errorf("Tests failed for location: %s | value: %s | received status: %t | received value: %s", test1.location, test1.value, test1.status, test1.retr)
-	}
+		Save(&(ele.location),&ele.value)									//Save Operations
+		ele.retr, ele.typedata, ele.status = Retrive(&(ele.location)) 		//Retrive Operations
 
-	if !(test2.value == test2.retr) || !test2.status {
-		t.Errorf("Tests failed for location: %s | value: %s | received status: %t | received value: %s", test2.location, test2.value, test2.status, test2.retr)
-	}
+		if (ele.value == ele.retr) && ele.status && (ele.typedata == ele.checkedData){
```
Comment (Harkishen-Singh) [inline gobase/plain_test.go]: indent space

File: tests/plain_test.go [hunk 5]
```diff
@@ -2,54 +2,39 @@ package tests
 
 import (
 	"testing"
-	"github.com/Harkishen-Singh/GO-file-database/gobase"
+	// "github.com/Harkishen-Singh/GO-file-database/gobase"
+	"../gobase"
```
Comment (Harkishen-Singh) [inline tests/plain_test.go]: alternate the comment, for semaphore tests to pass

File: tests/plain_test.go [hunk 6]
```diff
@@ -2,54 +2,39 @@ package tests
...
-	// retrive operations
 
-	test1.retr, test1.status = gobase.Retrive(test1.location)
-	test2.retr, test2.status = gobase.Retrive(test2.location)
-	test3.retr, test3.status = gobase.Retrive(test3.location)
-	test4.retr, test4.status = gobase.Retrive(test4.location)
+	for _,ele := range plainTests{
 
-	// checks
+		gobase.Save(&(ele.location),&ele.value)									//Save Operations
```
Comment (Harkishen-Singh) [inline tests/plain_test.go]: indent space

**Pull request merged**